src/python_testing/TC_CC_2_1.py

```diff
@@ -0,0 +1,419 @@
+            TestStep(
+                35, 'TH reads Primary4Y attribute from DUT and Verify that the DUT response contains an uint16 [Min:0 Max:0xfeff] if NumberOfPrimaries is 4 or more'),
+            TestStep(36, 'TH reads Primary4Intensity attribute from DUT and Verify that the DUT response contains an uint8 if NumberOfPrimaries is 4 or more'),
+            TestStep(
+                37, 'TH reads Primary5X attribute from DUT and Verify that the DUT response contains an uint16 [Min:0 Max:0xfeff] if NumberOfPrimaries is 5 or more'),
+            TestStep(
+                38, 'TH reads Primary5Y attribute from DUT and Verify that the DUT response contains an uint16 [Min:0 Max:0xfeff] if NumberOfPrimaries is 5 or more'),
+            TestStep(39, 'TH reads Primary5Intensity attribute from DUT and Verify that the DUT response contains an uint8 if NumberOfPrimaries is 5 or more'),
+            TestStep(
+                40, 'TH reads Primary6X attribute from DUT and Verify that the DUT response contains an uint16 [Min:0 Max:0xfeff] if NumberOfPrimaries is 6 or more'),
+            TestStep(
+                41, 'TH reads Primary6Y attribute from DUT and Verify that the DUT response contains an uint16 [Min:0 Max:0xfeff] if NumberOfPrimaries is 6 or more'),
+            TestStep(42, 'TH reads Primary6Intensity attribute from DUT and Verify that the DUT response contains an uint8 if NumberOfPrimaries is 6 or more'),
+            TestStep(43, 'TH reads from the DUT the (0x0030) WhitePointX attribute'),
+            TestStep(44, 'TH reads from the DUT the (0x0031) WhitePointY attribute'),
+            TestStep(45, 'TH reads from the DUT the (0x0032) ColorPointRX attribute'),
+            TestStep(46, 'TH reads from the DUT the (0x0033) ColorPointRY attribute'),
+            TestStep(47, 'TH reads from the DUT the (0x0034) ColorPointRIntensity attribute'),
+            TestStep(48, 'TH reads from the DUT the (0x0036) ColorPointGX attribute'),
+            TestStep(49, 'TH reads from the DUT the (0x0037) ColorPointGY attribute'),
+            TestStep(50, 'TH reads from the DUT the (0x0038) ColorPointGIntensity attribute'),
+            TestStep(51, 'TH reads from the DUT the (0x003a) ColorPointBX attribute'),
+            TestStep(52, 'TH reads from the DUT the (0x003b) ColorPointBY attribute'),
+            TestStep(53, 'TH reads from the DUT the (0x003c) ColorPointBIntensity attribute')
+        ]
+
+        return steps
+
+    async def _guard_and_read_attribute(self, attribute: str) -> Union[bool, Any]:
+        """Check using guard and reads the attribute value.
+        Args:
+            attribute (str): Attribute name from the cluster.
+        Returns:
+            Union[bool, Any]: Return the guard_status and value from the attribute if reached.
+        """
+        tmp_attr = None
+        attr_val = None
+        try:
```

Comment:
What is this for? Are you running into cases where you're trying to read attributes that are not part of the codegen?

Reply:
OH...ok, yeah, please don't pass in the atrribute name as a string. You can just use the attribute directly. That simplifies this function, and actually makes it unnecessary. You can just put the guard on the function below.